lib/lightning/usage_tracking/client.ex

```diff
@@ -9,14 +9,9 @@ defmodule Lightning.UsageTracking.Client do
   alias Lightning.UsageTracking.ResponseProcessor
 
   def submit_metrics(metrics, host) do
-    build_client(host)
-    |> post("/api/metrics", metrics)
-    |> elem(1)
-    |> ResponseProcessor.successful?()
-    |> then(fn
-      true -> :ok
-      false -> :error
-    end)
+    {_ok, env} = host |> build_client() |> post("/api/metrics", metrics)
+
+    if ResponseProcessor.successful?(env), do: :ok, else: :error
```

Comment:
When it fails the error tuple can have `any()`. I see the `successful?` handles it but I think it should not be called `env` when it can have another value (to make maintenance easier). 
A common Elixir way to handle this without the need for names is:
```
host
|> build_client()
|> post("/api/metrics", metrics)
|> case do
  {:ok, %Tesla.Env{status: 200}} -> :ok
  _error -> :error
end
```

Reply:
Thanks @jyeshe - I have updated the code in a new PR.